### DISCUSSION TITLE
hk1box emmc install

### DISCUSSION THREAD

**Discussion by mrnman66** [devmfc/debian-on-amlogic#28] [Q&A]
Hello,
 I have a tv box hk1box on soc s905x3. I want to install the image in emmc, but when I run the script I get errors. could you help me figure out what's wrong?
![photo1708105182](https://github.com/devmfc/debian-on-amlogic/assets/106779214/bb373a17-ffd3-4fb9-a80a-78417bfa8217)
![photo1708105123](https://github.com/devmfc/debian-on-amlogic/assets/106779214/48fb781f-b224-4514-afdf-8dbecccaa6a3)

**Comment by devmfc** [comment]
Did you use your USB flash drive also on another box with a different SOC? Try ‘source /etc/boot-run-once/03-set_fw_env_config‘ and try again.

**Reply by mrnman66** [reply]
Thank you very much, everything was installed and working. If I may ask one more question. Is KVM (Kernel-based Virtual Machine) enabled in the kernel?

**Reply by devmfc** [reply]
No, not at this moment (See /boot/config-<kernel version> for the used kernel config). Do you have a specific usecase for KVM on these 'not very powerful' ARM64 boxes?

**Reply by mrnman66** [reply]
I used hk1box with armbian image and proxmox 7. I installed several small virtual machines there, such as openwrt, home assistant os, it worked quite well if the operating system kernel supported KVM

**Reply by devmfc** [reply]
Really? I did not know there was an Arm64 version for Proxmox and I would never even think trying Proxmox on such boxes... Sounds interesting... I should try it sometime.

I will add KVM kernel support to the next release then.
(and in the mean time you can try attached kernel if you want. `dpkg -i linux-image-6.6.20-meson64_20240302_arm64.deb`)
[linux-image-6.6.20-meson64_20240302_arm64.deb.zip](https://github.com/devmfc/debian-on-amlogic/files/14472911/linux-image-6.6.20-meson64_20240302_arm64.deb.zip)
